lib/assets/traits/device_item_icon.dart

```diff
@@ -9,18 +9,31 @@ import 'package:yonomi_platform_sdk/yonomi-sdk.dart';
 
 class DeviceItemIcon {
   static Widget getIcon(List<Trait> traits) {
-    Trait determiningTrait = traits[0];
-    switch (determiningTrait.runtimeType) {
-      case LockTrait:
-        return LockIcon(determiningTrait.state.value);
-      case ThermostatTrait:
-        return ThermostatIcon(thermostatState: determiningTrait.state.value);
-      case PowerTrait:
-        return PowerItemIcon(determiningTrait.state.value);
-      case BatteryLevelTrait:
-        return BatteryLevelIcon(determiningTrait.state.value);
-      default:
-        return UnknownItemIcon();
+    try {
+      Trait determiningTrait = traits[0];
+      switch (determiningTrait.runtimeType) {
+        case LockTrait:
+          return LockIcon(determiningTrait.states
```

Comment:
Our pattern of firstWhere is probably dependent upon the query. We will need a separate class of how to determine dominantTrait. Maybe create a method for now which finds out the dominant trait for now